web/src/Storage.jsx

```diff
@@ -93,8 +99,40 @@ export default function Storage() {
     });
   }, [client.storage]);
 
+  useEffect(() => {
+    client.storage.getStatus().then(status => {
+      dispatch({ type: "CHANGE_STATUS", payload: status });
+    });
+  }, [client.storage]);
+
+  useEffect(() => {
+    return client.storage.onStatusChange(status => {
+      dispatch({ type: "CHANGE_STATUS", payload: status });
+    });
+  }, [client.storage]);
+
+  // FIXME: this useEffect should be removed after moving storage to its own service.
+  useEffect(() => {
+    client.manager.getStatus().then(status => {
+      dispatch({ type: "CHANGE_STATUS", payload: status });
+    });
+  }, [client.manager]);
+
+  // FIXME: this useEffect should be removed after moving storage to its own service.
```

Comment:
NP: maybe we should left a comment in the _backend_ part as a reminder...

Reply:
Actually, we do not need to do any changes in the backend related to this code. I would keep the comment just here.